### PR TITLE
Added a check for Visual Studio runtime at game start

### PR DESCRIPTION
**Pull request by Mailaender**
Another attempt to solve https://github.com/OpenRA/OpenRA/issues/8493.

### CODE REVIEW

**Comment by penev92** [comment]
OK, this seems to mostly work, with 2 comments:

File: OpenRA.GameMonitor/GameMonitor.cs [hunk 1]
```diff
@@ -15,16 +15,44 @@
...
+			{
+				var visualStudioRuntimeInstalled = 0;
+
+				foreach (var version in VisualStudioRuntimeVersions)
+				{
+					var key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\DevDiv\VC\Servicing\{0}\RuntimeMinimum\".F(version), false);
+					if (key != null && visualStudioRuntimeInstalled == 0)
+						visualStudioRuntimeInstalled = (int)key.GetValue("Install", 0);
+
+					key.Close();
```
Comment (penev92) [inline OpenRA.GameMonitor/GameMonitor.cs]: If you fail to open (find) the key, you get an NRE here. You do have a null check above though.

File: OpenRA.GameMonitor/GameMonitor.cs [hunk 2]
```diff
@@ -15,16 +15,44 @@
...
+				}
+
+				if (visualStudioRuntimeInstalled == 0)
+				{
+					var result = MessageBox.Show("OpenRA needs the Microsoft Visual C++ 2010 (or above) x86 Redistributable installed.",
+							"Visual C++ runtime not found.", MessageBoxButtons.OK, MessageBoxIcon.Error);
+
+					if (result == DialogResult.OK)
+						Process.Start("http://www.microsoft.com/en-us/download/details.aspx?id=5555");
```
Comment (penev92) [inline OpenRA.GameMonitor/GameMonitor.cs]: As a user, I really hate it when programs decide they can open random links whenever they feel like it. This is great to help people find what they need, but I'd rather it be optional.
Please replace `MessageBoxButtons.OK` above with `MessageBoxButtons.YesNo` and check if `result == Yes`.
(And add a "Do you want OpenRA to guide you to the download link now?" or something to the message.)

**Comment by pchote** [comment]
Doing this in our game (launcher) code At runtime is madness.  This kind of thing is what installers are for.

**Comment by Mailaender** [comment]
What I was asked for isn't possible in NSIS afaik, see https://github.com/OpenRA/OpenRA/pull/7609 but anyway go fix it yourself then.

**Comment by penev92** [comment]
The check was perfectly fine here. Actually made sure you didn't uninstall the prerequisites.
We even have a dedicated launcher for the occasion.

**Pull request closed without merging**